### ISSUE TITLE
Regression in v3.2.0 - when using null bytes in parameter strings.

### ISSUE THREAD

**Issue by samuelfekete** [pytest-dev/pytest#2669]
This error only happens when using the latest version: 3.2.0

When using a parameter string that contains a null character, for example:
```
@pytest.mark.parametrize(
    'display_name',
    ('null-byte-\0.txt', 'forward/slash', r'back\slash')
)
```
The following error is raised:
```
    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: must be string without null bytes, not str

.tox/py27/lib/python2.7/os.py:473: TypeError
```

**Comment by The-Compiler** [comment]
Duplicate of #2644, fixed by #2646, which will be part of pytest 3.2.1 soon.

**Comment by samuelfekete** [comment]
Thanks @The-Compiler
